backend/director/agents/dubbing.py

```diff
@@ -0,0 +1,172 @@
+        :param args: Additional positional arguments.
+        :param kwargs: Additional keyword arguments.
+        :return: The response containing information about the dubbing operation.
+        :rtype: AgentResponse
+        """
+        try:
+            self.videodb_tool = VideoDBTool(collection_id=collection_id)
+
+            # Get video audio file
+            video = self.videodb_tool.get_video(video_id)
+            if not video:
+                raise Exception(f"Video {video_id} not found")
+
+            if engine not in SUPPORTED_ENGINES:
+                raise Exception(f"{engine} not supported")
+
+            video_content = VideoContent(
+                agent_name=self.agent_name,
+                status=MsgStatus.progress,
+                status_message="Processing...",
+            )
+            self.output_message.content.append(video_content)
+            self.output_message.actions.append("Downloading video")
+            self.output_message.push_update()
+
+            download_response = self.videodb_tool.download(video["stream_url"])
+
+            os.makedirs(DOWNLOADS_PATH, exist_ok=True)
+            dubbed_file_path = f"{DOWNLOADS_PATH}/{video_id}_dubbed.mp4"
+
+            if engine == "elevenlabs":
+                ELEVENLABS_API_KEY = os.getenv("ELEVENLABS_API_KEY")
+                if not ELEVENLABS_API_KEY:
+                    raise Exception("Elevenlabs API key not present in .env")
+                elevenlabs_tool = ElevenLabsTool(api_key=ELEVENLABS_API_KEY)
+                job_id = elevenlabs_tool.create_dub_job(
+                    source_url=download_response["download_url"],
+                    target_language=target_language_code,
+                )
+                self.output_message.actions.append(
+                    f"Dubbing job initiated successfully - Job ID: {job_id}"
+                )
+                self.output_message.push_update()
+
+                self.output_message.actions.append(
+                    "Waiting for dubbing process to complete..."
+                )
+                self.output_message.push_update()
+                elevenlabs_tool.wait_for_dub_job(job_id)
+
+                self.output_message.actions.append("Downloading dubbed video")
+                self.output_message.push_update()
+                elevenlabs_tool.download_dub_file(
+                    job_id,
+                    target_language_code,
+                    dubbed_file_path,
+                )
+
+                self.output_message.actions.append(
+                    f"Uploading dubbed video to VideoDB as 'Dubbed_{target_language} {video['name']}'"
+                )
+                self.output_message.push_update()
+
+            dubbed_video = self.videodb_tool.upload(
+                dubbed_file_path,
+                source_type="file_path",
+                media_type="video",
+                name=f"Dubbed_{target_language} {video['name']}",
```

Comment:
Can we change it to either "{name} (Dubbed in {language)" or "(Dubbed) ({language}) {name}"?

Reply:
Changed it to [Dubbed in {target_language}] {video['name']}